modulemd/modulemd-module-stream-v1.c

```diff
@@ -2516,45 +2512,6 @@ modulemd_module_stream_v1_parse_module_components (
 }
 
 
-static GVariant *
-modulemd_module_stream_v1_parse_raw (yaml_parser_t *parser, GError **error)
-{
-  MODULEMD_INIT_TRACE ();
-  MMD_INIT_YAML_EVENT (event);
-  g_autoptr (GVariant) variant = NULL;
-  g_autoptr (GError) nested_error = NULL;
-
-  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
-
-  YAML_PARSER_PARSE_WITH_EXIT (parser, &event, error);
-
-  switch (event.type)
-    {
-    case YAML_SCALAR_EVENT:
-      variant =
-        mmd_variant_from_scalar ((const gchar *)event.data.scalar.value);
-      if (!variant)
-        {
-          MMD_YAML_ERROR_EVENT_EXIT (error, event, "Error parsing scalar");
-        }
-      break;
-
-    case YAML_MAPPING_START_EVENT:
-      variant = mmd_variant_from_mapping (parser, &nested_error);
-      break;
-
-    default:
-      MMD_YAML_ERROR_EVENT_EXIT (error,
-                                 event,
-                                 "Unexpected YAML event in raw parsing: %s",
-                                 mmd_yaml_get_event_name (event.type));
-      break;
-    }
-
-  return g_steal_pointer (&variant);
```

Comment:
I'm not familiar with `GVariant`s. I see v1 was returning `g_steal_pointer (&variant)` while v2 (and the v3 clone and your new common function) return `g_variant_ref_sink (variant)`. Are they pretty much the same?

Reply:
In this particular instance, they are the same. `g_variant_ref_sink()` is more correct here, but they essentially end up compiling down to the same code.

Reply:
Put another way: we should probably fix it for v1, but since v1 is dead and buried it likely doesn't really matter.

Reply:
Thanks for the explanation. I assume by "v1" you are referring to libmodulemd V1, since this PR now fixes if for _stream_ v1 in libmodulemd V2.